### DISCUSSION TITLE
filter out the 'swift' tag

### DISCUSSION THREAD

**Discussion by tayloraswift** [SwiftPackageIndex/SwiftPackageIndex-Server#1959] [Ideas]
some of us package authors use the [`'swift'`](https://swiftpackageindex.com/keywords/swift) tag on github so our packages show up in searches for swift packages for a particular domain. but every package on the swift package index is a swift (5) package. 

![image](https://user-images.githubusercontent.com/2556986/183509691-da4ffa8c-efee-40bb-b42a-9d302b3e025a.png)

maybe add a rule to filter out `'swift'`, `'swift5'`, and other meaningless tags?

**Comment by Sherlouk** [comment]
I agree to be honest, there are a number of very noisy keywords like, as you say, Swift but also ones like swift-package-manager etc.

I should hope so if it's on the Swift Package Index 😂

Doesn't feel beneficial to surface these to users in the list

**Comment by Sherlouk** [comment]
I wonder if it could also relate to the search changes to stop these noisy keywords from affecting rankings?

**Reply by heckj** [reply]
There's a tradeoff in the search mechanisms where the noisy bits that happen to be patterns within other words are getting exposed. The gist of that trade-off is that if we normalize the terms using full-text search mechanisms (stemming to get lexemes, including synonym dictionaries and filtered stopwords), AND we use that to determine if search results will appear at all, then we immediately dump out any results from non-english characters, and rather heavily impact jargon-specific terms. 

For example, in the search updates, we tried that technique (using English as the language du-jour), but there are Mandarin summary elements that then immediately drop from the search results, never to be found. In addition, the stemmers and lexeme systems couldn't differentiate NS and UI as prefixes that can/should be unrolled from terms to get to a root lexeme. So "UIBezierPath" wouldn't match with "bezier" as a search term.

The route we ended up taking (you can see all the experimentation and detail in https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/discussions/1876) is to compute the English language terms but ONLY use them as a means to generate a ranking value, and then present all the found results in the order of that ranking. This has the effect of popping the sub-terms (if you were searching for `ping`, `ftp`, or `swift` above the other terms where they're "included within the strings of characters" for larger terms, potentially unrelated. 

It was the optimal balance of improving the search results without the dropped downsides. So there's still can be a lot of irrelevant results, but they should be ranked below the ones that appear to be more relevant.
